src/main/java/racingcar/model/RacingCar.java

```diff
@@ -0,0 +1,59 @@
+package racingcar.model;
+
+import racingcar.model.movestrategy.MoveStrategy;
+import racingcar.model.movestrategy.RandomVarMoveStrategy;
+
+public class RacingCar {
+    private static final String ERR_NAME_LENGTH = "Names longer than 5 characters are not allowed.";
+    private static final int NAME_MAX_LENGTH = 5;
+
+    private MoveStrategy moveStrategy;
+    private Position position;
+    private String name;
+
+    public RacingCar(String name) {
+        // set MoveStrategy
+        this.moveStrategy = new RandomVarMoveStrategy();
+
+        // validation
+        validateCarName(name);
+
+        // setting
+        this.name = name;
+        this.position = new Position();
+    }
+
+    public RacingCar(MoveStrategy moveStrategy, String name) {
+        // set MoveStrategy
+        this.moveStrategy = moveStrategy;
+
+        // validation
+        validateCarName(name);
+
+        // setting
+        this.name = name;
+        this.position = new Position();
+    }
+
+    private void validateCarName(String name) {
+        if (name.length() > NAME_MAX_LENGTH) {
+            throw new IllegalArgumentException(ERR_NAME_LENGTH);
+        }
+    }
+
+
+    public void moveOrStop(int value) {
+        if (moveStrategy.canMove(value)) {
+            this.position.step();
+        }
+    }
```

Comment:
moveStrategy는 moveOrStop 메서드에서만 사용되는데요
필드로 관리하기보다 메서드의 인자로 받으면 어떨까요?
```suggestion
    public void moveOrStop(MoveStrategy moveStrategy) {
        if (moveStrategy.canMove()) {
            this.position.step();
        }
    }
```

Reply:
제 생각에는 1회의 racing game 안에서 각각의 racing car들이 하나의 move strategy로 움직여야 한다고 생각해서 move strategy를 car class의 필드로 두었습니다!
수정없이 진행하는 것에 대해서 어떻게 생각하시는지 궁금합니다! 
혹시 제가 생각하지 못한 부분이 있을까요??